libs/parallelism/algorithms/include/hpx/algorithms/traits/projected.hpp

```diff
@@ -100,6 +102,34 @@ namespace hpx { namespace parallel { namespace traits {
         {
         };
 #endif
+
+#if defined(HPX_HAVE_CXX20_EXPERIMENTAL_SIMD)
+        // This is being instantiated if a simd pack execution policy is used
+        // with a zip_iterator. In this case the function object is invoked
+        // with a tuple<datapar<T>...> instead of just a tuple<T...>
+        template <typename Proj, typename ValueType, typename Enable = void>
+        struct projected_result_of_simd_pack_
+          : hpx::util::invoke_result<Proj,
+                typename hpx::parallel::traits::simd_pack_load<
+                    typename hpx::parallel::traits::simd_pack_type<
+                        ValueType>::type,
+                    ValueType>::value_type&>
+        {
+        };
+
+        template <typename Projected, typename Enable = void>
+        struct projected_result_of_simd_pack;
+
+        template <typename Projected>
+        struct projected_result_of_simd_pack<Projected,
+            typename hpx::util::always_void<
+                typename Projected::iterator_type>::type>
+          : projected_result_of_simd_pack_<typename Projected::projector_type,
+                typename std::iterator_traits<
+                    typename Projected::iterator_type>::value_type>
+        {
+        };
+#endif
```

Comment:
Same here. Let's try to specialize the `projected_result_of_vector_pack` trait (that's what it is for) instead of creating a new trait just for the `simd` types.